tests/qos/test_buffer.py

```diff
@@ -1829,15 +1884,31 @@ def test_buffer_model_test(duthosts, rand_one_dut_hostname, conn_graph_facts):
 
 def calculate_headroom_data(duthost, port_to_test):
     """
-    This function is intend to calculate the headroom size based on the unput port attributes(speed, cable_length, number of lanes..., etc)
+    This function is intend to calculate the headroom size based on the unput port attributes
+    Each vendor should have it's own implementation for the algorithm
+    """
+    if ASIC_TYPE == 'mellanox':
+        return mellanox_calculate_headroom_data(duthost, port_to_test)
+    else:
+        return False, None
+
+
+def mellanox_calculate_headroom_data(duthost, port_to_test):
+    """
+    This function is Mellanox platform specific.
+    It intends to calculate the headroom size based on the unput port attributes(speed, cable_length, number of lanes..., etc)
     This algorithm is the same as the implementation in https://github.com/Azure/sonic-swss/blob/master/cfgmgr/buffer_headroom_mellanox.lua
     """
-    lossless_mtu = 9100
-    small_packet_percentage = 50
+    global ASIC_TABLE_KEYS_LOADED
+    global cell_size
```

Comment:
Better to use the upper case for global variables. There was a comment from the community about that.